internal/pkg/scaffold/cmd.go

```diff
@@ -122,6 +123,7 @@ func main() {
 	// Create a new Cmd to provide shared dependencies and start components
 	mgr, err := manager.New(cfg, manager.Options{
 		Namespace:          namespace,
+		MapperProvider:     restmapper.NewDynamicRESTMapper,
```

Comment:
I am wondering if we should make this an opt-in feature because missing there is no mechanism's in place to back off expensive reloads on every miss. 

Thoughts?

Reply:
I reasoned that it's better to be slow than wrong, by default, but I could go either way.

Reply:
Solly wanted a backoff upstream (controller-runtime); that's why I haven't just submitted this there already.

I guess the case where it would matter would be if you're waiting for another operator to create a bunch of objects of some CRD type. So then every now and then you do 50 GETs, to see if the 50 objects have been created. Which, with this code, could result in 50 restmapper reloads if the CRD hadn't been created yet.

Reply:
I'm probably missing something but do we only incur the miss when the client/cache is used to CRUD an instance/CR of a newly created CRD whose mapping is not present in the DynamicRestMapper?

If so then I assume most likely this call(e.g GET) would be made inside the reconcile loop. If the DynamicRestMapper reloads and misses(because the CRD is not yet created), and we get an error from `client.Get()`, then we can just rely on backoff of the controller's workqueue right?
https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/internal/controller/controller.go#L210-L215

If this call is not in the reconcile loop, then I imagine most likely the operator exits and restarts in the absence of a retry.